extensions/amp-kaltura-player/0.1/amp-kaltura-player.js

```diff
@@ -0,0 +1,125 @@
+
+    /** @override */
+    isLayoutSupported(layout) {
+      return isLayoutSizeDefined(layout);
+    }
+
+    /** @override */
+    buildCallback() {
+      if (!this.getPlaceholder()) {
+        this.buildImagePlaceholder_();
+      }
+    }
+
+    /** @override */
+    layoutCallback() {
+      const width = this.element.getAttribute('width');
+      const height = this.element.getAttribute('height');
+      const partnerid = AMP.assert(
+        this.element.getAttribute('data-partner'),
+        'The data-account attribute is required for <amp-kaltura-player> %s',
+        this.element);
+      const uiconfid = this.element.getAttribute('data-uiconf') ||
+      this.element.getAttribute('data-uiconf-id') ||
+        'default';
+      const entryid = this.element.getAttribute('data-entryid') || 'default';
+      const iframe = document.createElement('iframe');
+      let src = `https://cdnapisec.kaltura.com/p/${encodeURIComponent(partnerid)}/sp/${encodeURIComponent(partnerid)}00/embedIframeJs/uiconf_id/${encodeURIComponent(uiconfid)}/partner_id/${encodeURIComponent(partnerid)}?iframeembed=true&playerId=kaltura_player_amp&entry_id=${encodeURIComponent(entryid)}`;
+      const params = {};
+
+        // Pass through data-param-* attributes as params for plugin use
+      for (let i = 0; i < this.element.attributes.length; i++) {
+        const attr = this.element.attributes[i];
+        const matches = attr.nodeName.match(/^data-param-(.+)/);
+        if (matches) {
+          const param = dashToCamelCase(matches[1]);
+          const flashvar = 'flashvars[' + param + ']';
+          params[flashvar] = attr.nodeValue;
+        }
+      }
+
+      src = addParamsToUrl(src, params);
+      iframe.setAttribute('frameborder', '0');
+      iframe.setAttribute('allowfullscreen', 'true');
+      iframe.src = src;
+      this.applyFillContent(iframe);
+      iframe.width = width;
+      iframe.height = height;
+      this.element.appendChild(iframe);
+        /** @private {?Element} */
+      this.iframe_ = iframe;
+      return loadPromise(iframe);
+    }
+
+    /** @private */
+    buildImagePlaceholder_() {
+      const imgPlaceholder = new Image();
+
+      setStyles(imgPlaceholder, {
+            //Cover matches YouTube Player styling.
+        'object-fit': 'cover',
+            //Hiding the placeholder initially to give the browser time to fix
+            // the object-fit: cover.
+        'visibility': 'hidden',
+      });
+      const width = this.element.getAttribute('width');
+      const height = this.element.getAttribute('height');
+      const partnerid = AMP.assert(
+        this.element.getAttribute('data-partner'),
+        'The data-account attribute is required for <amp-kaltura-player> %s',
+        this.element);
+      const entryid = this.element.getAttribute('data-entryid') || 'default';
+      let src = `https://cdnapisec.kaltura.com/p/${encodeURIComponent(partnerid)}/thumbnail/entry_id/${encodeURIComponent(entryid)}`;
```

Comment:
Just making sure: Is this the exact same URL the player will load for its poster?